src/org/labkey/idri/query/ConcentrationsTable.java

```diff
@@ -38,38 +43,41 @@
             FieldKey.fromString("Concentration"),
             FieldKey.fromString("Unit")};
 
-    public ConcentrationsTable(idriSchema schema)
+    ConcentrationsTable(idriSchema schema, ContainerFilter cf)
     {
-        super(schema.getDbSchema().getTable(idriSchema.TABLE_CONCENTRATIONS), schema);
+        super(schema.getDbSchema().getTable(idriSchema.TABLE_CONCENTRATIONS), schema, cf);
 
         addColumn(wrapColumn("rowId", getRealTable().getColumn("rowId")));
-        ColumnInfo compoundCol = addColumn(wrapColumn("Compound", getRealTable().getColumn("Compound")));
-        compoundCol.setFk(new LookupForeignKey("rowid")
+        BaseColumnInfo compoundCol = addColumn(wrapColumn("Compound", getRealTable().getColumn("Compound")));
+        // TODO: Replace this LookupForeignKey with QueryForeignKey.from() (which will allow caching, etc.)
+        compoundCol.setFk(new LookupForeignKey(cf, "rowid", null)

```

Comment:
can we use the same lookupFK instance for each of these?

Reply:
Good idea..  Should do, but not necessary for merge I think.